modules/dns_common/dns_common.rb

```diff
@@ -24,5 +24,29 @@ def dns_find key
     rescue Resolv::ResolvError
       false
     end
+
+    def ptr_to_ip ptr
+     if ptr =~ /\.in-addr\.arpa$/
+       ptr.split('.')[0..-3].reverse.join('.')
+     elsif ptr =~ /\.ip6\.arpa$/
+       ptr.split('.')[0..-3].reverse.each_slice(4).inject([]) {|address, word| address << word.join}.join(":")
+     else
+       raise Proxy::Dns::Error.new("Not a PTR record: '#{ptr}'")
+     end
+    end
+
+    def a_record_conflicts?(fqdn, ip)
+      addresses = resolver.getaddresses(fqdn)
+      addresses.each { |address| return address.to_s if address.to_s.casecmp(ip) == 0 }
```

Comment:
Dejavu: I had to lookup casecmp yesterday because I did not know about it's returns; should have known you would suggest the same thing ;)